parse/lex.go

```diff
@@ -98,11 +100,16 @@ func (l *Lexer) NextToken() Token {
 // state has the input(codes) as a string and has the current position and the line.
 type state struct {
 	input string
-	start int
-	pos   int
+	start Pos
+	pos   Pos
```

Comment:
I think that `pos` name would be more specific because type `Pos` might confuse a variable name `pos` with type `Pos`.

Reply:
I changed pos to end!!